File: python-cli/src/convex_notes/ui/banner.py
"""Application banner."""

from rich.console import Console
from rich.panel import Panel
from rich.text import Text

console = Console()


def print_banner() -> None:
    """Print the application banner."""
    title = Text()
    title.append("📝 CONVEX NOTES MANAGER", style="bold white")

    subtitle = Text()
    subtitle.append("Self-Hosted • Python Client • v0.1.0", style="dim")

    content = Text.assemble(title, "\n", subtitle)

    panel = Panel(
        content,
        border_style="cyan",
        padding=(1, 4),
    )

    console.print()
    console.print(panel)
    console.print()

File: bun-cli/src/ui/banner.ts
/**
 * Banner UI component.
 */

import pc from "picocolors";

export function printBanner(): void {
  console.log();
  console.log(
    pc.cyan("╔══════════════════════════════════════════════════════════╗"),
  );
  console.log(
    pc.cyan("║") +
      pc.bold(
        pc.white("           📝 CONVEX NOTES MANAGER                      "),
      ) +
      pc.cyan("║"),
  );
  console.log(
    pc.cyan("║") +
      pc.dim("         Self-Hosted • Bun Client • v0.1.0             ") +
      pc.cyan("║"),
  );
  console.log(
    pc.cyan("╚══════════════════════════════════════════════════════════╝"),
  );
  console.log();
}
